### ISSUE TITLE
Smoke test failure: test_vm_life_cycle.py 

### ISSUE THREAD

**Issue by weizhouapache** [apache/cloudstack#7786]
The following failures happened in the trillian test of many PRs

Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_01_migrate_VM_and_root_volume | `Error` | 80.81 | test_vm_life_cycle.py
test_02_migrate_VM_with_two_data_disks | `Error` | 53.77 | test_vm_life_cycle.py

The test run on kvm-centos7 environment.

logs shows

```
======================================================================
ERROR: Test VM will be migrated with it's root volume
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/marvin/tests/smoke/test_vm_life_cycle.py", line 1628, in test_01_migrate_VM_and_root_volume
    response = self.apiclient.migrateVirtualMachineWithVolume(cmd)
  File "/usr/local/lib/python3.6/site-packages/marvin/cloudstackAPI/cloudstackAPIClient.py", line 3014, in migrateVirtualMachineWithVolume
    response = self.connection.marvinRequest(command, response_type=response, method=method)
  File "/usr/local/lib/python3.6/site-packages/marvin/cloudstackConnection.py", line 381, in marvinRequest
    raise e
  File "/usr/local/lib/python3.6/site-packages/marvin/cloudstackConnection.py", line 376, in marvinRequest
    raise self.__lastError
  File "/usr/local/lib/python3.6/site-packages/marvin/cloudstackConnection.py", line 105, in __poll
    % async_response)
Exception: Job failed: {accountid : '43711834-2c2c-11ee-9391-1e00ef0003de', userid : '43720e4e-2c2c-11ee-9391-1e00ef0003de', cmd : 'org.apache.cloudstack.api.command.admin.vm.MigrateVirtualMachineWithVolumeCmd', jobstatus : 2, jobprocstatus : 0, jobresultcode : 530, jobresulttype : 'object', jobresult : {errorcode : 530, errortext : 'Failed to migrate VM [VM instance {"id":346,"instanceName":"i-228-346-VM","type":"User","uuid":"a23d5b1d-f86e-45bf-9c90-4aefc3bd4c04"}] along with its volumes due to [com.cloud.utils.exception.CloudRuntimeException: Copy volume(s) to storage(s) [{volume: "398", from: "2", to:"1"}] and VM to host [{vm: "346", from: "2", to:"1"}] failed in StorageSystemDataMotionStrategy.copyAsync. Error message: [Exception during migrate: org.libvirt.LibvirtException: internal error: unable to execute QEMU command \'migrate\': this feature or command is not currently supported].].'}, jobinstancetype : 'VirtualMachine', jobinstanceid : 'a23d5b1d-f86e-45bf-9c90-4aefc3bd4c04', created : '2023-07-27T12:29:13+0000', completed : '2023-07-27T12:29:17+0000', jobid : '1e846d8b-830f-4e0e-8b47-c2d32b9d3143'}
```
and
```
======================================================================
ERROR: Test VM will be migrated with it's root volume
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/marvin/tests/smoke/test_vm_life_cycle.py", line 1669, in test_02_migrate_VM_with_two_data_disks
    response = self.apiclient.migrateVirtualMachineWithVolume(cmd)
  File "/usr/local/lib/python3.6/site-packages/marvin/cloudstackAPI/cloudstackAPIClient.py", line 3014, in migrateVirtualMachineWithVolume
    response = self.connection.marvinRequest(command, response_type=response, method=method)
  File "/usr/local/lib/python3.6/site-packages/marvin/cloudstackConnection.py", line 381, in marvinRequest
    raise e
  File "/usr/local/lib/python3.6/site-packages/marvin/cloudstackConnection.py", line 376, in marvinRequest
    raise self.__lastError
  File "/usr/local/lib/python3.6/site-packages/marvin/cloudstackConnection.py", line 105, in __poll
    % async_response)
Exception: Job failed: {accountid : '43711834-2c2c-11ee-9391-1e00ef0003de', userid : '43720e4e-2c2c-11ee-9391-1e00ef0003de', cmd : 'org.apache.cloudstack.api.command.admin.vm.MigrateVirtualMachineWithVolumeCmd', jobstatus : 2, jobprocstatus : 0, jobresultcode : 530, jobresulttype : 'object', jobresult : {errorcode : 530, errortext : 'Failed to migrate VM [VM instance {"id":348,"instanceName":"i-228-348-VM","type":"User","uuid":"a19c3074-f6dd-42f0-807e-96aa211c3eaa"}] along with its volumes due to [com.cloud.utils.exception.CloudRuntimeException: Copy volume(s) to storage(s) [{volume: "401", from: "2", to:"2"},{volume: "402", from: "2", to:"1"},{volume: "403", from: "1", to:"2"}] and VM to host [{vm: "348", from: "2", to:"1"}] failed in StorageSystemDataMotionStrategy.copyAsync. Error message: [Exception during migrate: org.libvirt.LibvirtException: internal error: unable to execute QEMU command \'migrate\': this feature or command is not currently supported].].'}, jobinstancetype : 'VirtualMachine', jobinstanceid : 'a19c3074-f6dd-42f0-807e-96aa211c3eaa', created : '2023-07-27T12:30:06+0000', completed : '2023-07-27T12:30:11+0000', jobid : 'bfbf8a7e-63fe-4433-88cf-3f7cc1696233'}
```

##### ISSUE TYPE

 * Bug Report
 * Improvement Request
 * Enhancement Request
 * Feature Idea
 * Documentation Report
 * Other

##### COMPONENT NAME

~~~

~~~

##### CLOUDSTACK VERSION

~~~
4.18
~~~

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~

~~~

##### ACTUAL RESULTS

~~~

~~~

**Comment by DaanHoogland** [comment]
@weizhouapache `org.libvirt.LibvirtException: internal error: unable to execute QEMU command \'migrate\': this feature or command is not currently supported` means the environment doesn't support the test.
We can only make the test more aware of the libvirt version and mark it as skipped istead of failed, I am afraid.

**Comment by weizhouapache** [comment]
> @weizhouapache `org.libvirt.LibvirtException: internal error: unable to execute QEMU command \'migrate\': this feature or command is not currently supported` means the environment doesn't support the test. We can only make the test more aware of the libvirt version and mark it as skipped istead of failed, I am afraid.

@DaanHoogland 
did the tests work on centos7 before ? were they broken by some changes ?

**Comment by DaanHoogland** [comment]
> > @weizhouapache `org.libvirt.LibvirtException: internal error: unable to execute QEMU command \'migrate\': this feature or command is not currently supported` means the environment doesn't support the test. We can only make the test more aware of the libvirt version and mark it as skipped istead of failed, I am afraid.
> 
> @DaanHoogland did the tests work on centos7 before ? were they broken by some changes ?

the change (i think) is in the test env, not in the code. Qemu/libvirt is updated to a version that no longer supports live migration.

**Comment by weizhouapache** [comment]
> > > @weizhouapache `org.libvirt.LibvirtException: internal error: unable to execute QEMU command \'migrate\': this feature or command is not currently supported` means the environment doesn't support the test. We can only make the test more aware of the libvirt version and mark it as skipped istead of failed, I am afraid.
> > 
> > 
> > @DaanHoogland did the tests work on centos7 before ? were they broken by some changes ?
> 
> the change (i think) is in the test env, not in the code. Qemu/libvirt is updated to a version that no longer supports live migration.

@DaanHoogland 
yes, could be

we could drop centos7 as kvm in testing. rocky8 or ubuntu22 can be default.

**Comment by DaanHoogland** [comment]
> > > > @weizhouapache `org.libvirt.LibvirtException: internal error: unable to execute QEMU command \'migrate\': this feature or command is not currently supported` means the environment doesn't support the test. We can only make the test more aware of the libvirt version and mark it as skipped istead of failed, I am afraid.
> > > 
> > > 
> > > @DaanHoogland did the tests work on centos7 before ? were they broken by some changes ?
> > 
> > 
> > the change (i think) is in the test env, not in the code. Qemu/libvirt is updated to a version that no longer supports live migration.
> 
> @DaanHoogland yes, could be
> 
> we could drop centos7 as kvm in testing. rocky8 or ubuntu22 can be default.

let's test vm life cycle in those (rocky8++ and ubuntu)

**Comment by weizhouapache** [comment]
> > 
> > @DaanHoogland yes, could be
> > we could drop centos7 as kvm in testing. rocky8 or ubuntu22 can be default.
> 
> let's test vm life cycle in those (rocky8++ and ubuntu)

@DaanHoogland 
on rocky8 and ubuntu22, all are fine.
these failures only happened on kvm-centos7 environments.

**Comment by weizhouapache** [comment]
Fixed by #7862
